#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-catch-all.dpatch by Tzafrir Cohen <tzafrir.cohen@xorcom.com>
##
## DP: Catch-all should also catch a DID of 1 digit.

@DPATCH@
diff -urNad upstream~/amp_conf/htdocs/admin/modules/core/functions.inc.php upstream/amp_conf/htdocs/admin/modules/core/functions.inc.php
--- upstream~/amp_conf/htdocs/admin/modules/core/functions.inc.php	2007-06-08 03:42:50.000000000 +0300
+++ upstream/amp_conf/htdocs/admin/modules/core/functions.inc.php	2007-07-31 15:32:29.000000000 +0300
@@ -205,6 +205,14 @@
 							$catchall = true;
 						$ext->add($catchall_context, $catchaccount, '', new ext_NoOp('Catch-All DID Match - Found ${EXTEN} - You probably want a DID for this.'));
 						$ext->add($catchall_context, $catchaccount, '', new ext_goto('1','s','ext-did'));
+		
+						// this is for catching a route with only one digit.
+                                                if ($catchall)
+                                                {
+							$ext->add($catchall_context, "_X", '', new ext_NoOp('Catch-All DID Match - Found ${EXTEN} - You probably want a DID for this.'));
+							$ext->add($catchall_context, "_X", '', new ext_goto('1','s','ext-did'));
+						}
+
 					}
 					
 					if ($item['faxexten'] != "default") {
